client/signup/steps/site-topic/index.jsx

```diff
@@ -26,23 +28,36 @@ class SiteTopicStep extends Component {
 		setSiteTopic: PropTypes.func.isRequired,
 		signupProgress: PropTypes.array,
 		stepName: PropTypes.string,
+		siteTopic: PropTypes.string,
 	};
 
 	state = {
-		siteTopic: '',
+		siteTopicInputValue: '',
 	};
 
-	onChangeTopic = event => this.setState( { siteTopic: event.target.value } );
+	componentDidMount() {
+		const { siteTopic } = this.props;
 
-	// TODO:
-	// Handle submission.
-	submitSiteTopic( event ) {
-		event.preventDefault();
+		if ( siteTopic ) {
+			this.state.siteTopicInputValue = siteTopic;
+		}
 	}
 
+	onChangeTopic = event => this.setState( { siteTopicInputValue: event.target.value } );
+
+	submitSiteTopic = event => {
+		event.preventDefault();
+
+		const { goToNextStep, flowName } = this.props;
+
+		this.props.setSiteTopic( this.state.siteTopicInputValue );
```

Comment:
I think we might need here `SignupActions.submitSignupStep()` to add to the dependency store? Or are you planning it for a subsequent PR?

Reply:
I asked this exact question to @michaeldcain :)- what I understood is that the dependency store is more to check if each step provides its dependencies and if the final API can make its call with all dependencies in place, while the redux store is from where the values are fetched. This redundancy is due to legacy reasons. There was an earlier attempt to move to a single source of truth in signup, but there were some issues. So, there is no imminent plan to move to a single source of truth in the next three months.

Reply:
Awesome. Thanks for the clarification :)